client/client.go

```diff
@@ -2167,6 +2198,20 @@ func (c *Client) allocSync() {
 				continue
 			}
 
+			c.allocLock.RLock()
+			for _, update := range sync {
+				if ar, ok := c.allocs[update.ID]; ok {
+					ar.AcknowledgeState(&arstate.State{
```

Comment:
If you wanted to avoid storing a bunch of duplicated client alloc state and implementing a deep equality, I think we could make this a `uint64`, increment it on every `Client.AllocStateUpdated` call, and have the Client persist it here.

That relies on ARs/TRs not calling `Client.AllocStateUpdated` even when nothing has been updated though, so I think your approach is safer.